src/mcp/server/streamable_http.py

```diff
@@ -559,8 +560,9 @@ async def _handle_get_request(self, request: Request, send: Send) -> None:
             await response(request.scope, request.receive, send)
             return
 
-        if not await self._validate_session(request, send):
+        if not await self._validate_request_headers(request, send):
```

Comment:
Do we need to test this too? (and propagate the version in the headers sent by sse_client)

Reply:
This is actually an interesting point 🤔 while we have the headers for version in a few places, we don't actually seem to _do_ anything with it on the server as far as I can tell (at least in the Python SDK) other than make sure that the version is supported.

I don't think we modify function behavior based on protocol version anywhere, so I'm not actually sure where (if anywhere) we would propagate this information to.

If we had the server respond with a version header it would make sense to jam it into the response object, but because we don't I don't see where we should propagate it to 🤔 

Maybe @ihrpr or @bhosmer-ant have a thought here?